src/meltano/core/db.py

```diff
@@ -41,8 +42,9 @@ def project_engine(
     settings = ProjectSettingsService(project)
 
     engine_uri = settings.get("database_uri")
-    logging.debug(f"Creating engine {project}@{engine_uri}")
-    engine = create_engine(engine_uri, pool_pre_ping=True)
+    logging.debug(f"Creating engine '{project}@{engine_uri}'")
+
+    engine = create_engine(engine_uri, poolclass=NullPool)
```

Comment:
I want to call attention to this change. Using a `NullPool` seems fine for Meltano, since we don't need connection pooling to the best of my knowledge.

Using connection pooling was causing some tricky issues with the tests. It was introduced when we added support for Postgres as the DB backend. If we don't have a good reason for using connection pooling, I'm inclined to not use it (i.e. use `NullPool`) to simplify it, and in the process, resolve this Heisenbug that I've been dealing with for the past 48 hours

Reply:
I think this is ok as i) connection pooling is an optimisation for applications with 100's or 1000's of users spawning connections, which our app isn't and ii) because we aren't going to be using Postgres (or any RDBMS) for much longer 🙂